src/test/util/setup_common.cpp

```diff
@@ -144,6 +144,7 @@ BasicTestingSetup::BasicTestingSetup(const ChainType chainType, const std::vecto
 
 BasicTestingSetup::~BasicTestingSetup()
 {
+    m_node.kernel.reset();
```

Comment:
In commit "tests: Reset node context members on ~ChainTestingSetup" (960112dbb1fb028b73b6e83932dd457f5a69ab5d)

I think this change makes sense but the log message is misleading because in the Shutdown() function the kernel context is currently destroyed _before_ chainman, mempool, scheduler, not after:

https://github.com/bitcoin/bitcoin/blob/d18a8f6f6969487021b8fd50391a2a8d1dc29844/src/init.cpp#L346-L350

The test code is more correct than the Shutdown() code because ~BasicTestingSetup runs after ~ChainTestingSetup, and the chainman and scheduler objects should be destroyed before kernel context because they might rely on it.

It'd be good to either clarify the log message in this commit to say mention this part of the change doesn't match Shutdown function, or update the Shutdown function to destroy the kernel context last like the test. (I could also update the Shutdown function in #28051 if you want this PR to be test-only)